lib/Common/DataStructures/DictionaryEntry.h

```diff
@@ -118,11 +118,23 @@ namespace JsUtil
         TKey const& Key() const  { return key; }
     };
 
+    template <class TKey>
+    class BaseKeyEntry : public ValueEntry<TKey>
```

Comment:
Why have this base class? Can we just have KeyEntry?

Reply:
Honestly? I was following the style of `KeyValueEntry` which had a base class. If you think it'd be better to remove the additional layer I'd be happy to.